src/chi-vue/src/components/data-table/DataTable.tsx

```diff
@@ -0,0 +1,852 @@
+        ? DATA_TABLE_CLASSES.ROW_CHILD
+        : DATA_TABLE_CLASSES.ROW;
+
+    if (this._expandable) {
+      if (bodyRow.nestedContent) {
+        rowCells.push(this.expansionButton(rowId));
+        rowAccordionContent.push(this.rowAccordionContent(bodyRow.nestedContent));
+      } else {
+        rowCells.push(
+          <div
+            class={`${DATA_TABLE_CLASSES.CELL} ${DATA_TABLE_CLASSES.EXPANDABLE}`}
+            role="cell"
+          />
+        );
+      }
+    }
+
+    if (this.config.selectable) {
+      rowCells.push(this.selectRowCheckbox(false, rowId));
+    }
+
+    let cellIndex = 0;
+
+    bodyRow.data.forEach((rowCell: any) => {
+      const columnSettings = this.data.head[
+          Object.keys(this.data.head)[cellIndex]
+        ],
+        alignment = this.cellAlignment(
+          rowCell.align ? rowCell.align : columnSettings.align || null
+        ),
+        cellLabel = rowCell.label || columnSettings.label,
+        cellWidth =
+          this.config.columnSizes && this._currentScreenBreakpoint
+            ? this.config.columnSizes[this._currentScreenBreakpoint][
+                cellIndex
+              ]
+            : null;
+      let cellData;
+
+      if (!!rowCell.template && !!this.$scopedSlots[rowCell.template]) {
+        if (typeof rowCell === 'object' && rowCell.payload) {
+          cellData = this.$scopedSlots[rowCell.template]!(rowCell.payload);
+        }
+      } else if (typeof rowCell === 'object' && !!rowCell.value) {
+        cellData = (
+          <Tooltip message={rowCell} class="-w--100">
+            <div class={UTILITY_CLASSES.TYPOGRAPHY.TEXT_TRUNCATE}>
+              {rowCell.value}
+            </div>
+          </Tooltip>
+        );
+      } else if (
+        typeof rowCell === 'string' ||
+        typeof rowCell === 'number') {
+        cellData = (
+          <Tooltip message={rowCell} class="-w--100">
+            <div class={UTILITY_CLASSES.TYPOGRAPHY.TEXT_TRUNCATE}>
+              {rowCell}
+            </div>
+          </Tooltip>
+        );
+      } else {
+        cellData = null;
+      }
+
+      rowCells.push(
+        <div
+          aria-label={cellLabel}
+          class={`
+            ${DATA_TABLE_CLASSES.CELL}
+            ${alignment}
+            ${cellWidth && cellWidth > 0 ? `-flex-basis--${cellWidth}` : ''}`}
+          data-label={cellLabel}
+          role="cell"
+          style={`
+            ${cellWidth === 0 ? 'display: none' : ''}
+            ${columnSettings.allowOverflow ? 'overflow: visible;' : ''}
+            `}
+        >
+          {cellData}
+        </div>
+      );
+
+      cellIndex++;
+    });
+
+    row.push(
+      <div
+        id={rowId}
+        class={`
+        ${rowClass}
+        ${striped ? DATA_TABLE_CLASSES.STRIPED : ''}
+        ${this.selectedRows.includes(rowId) ? ACTIVE_CLASS : ''}
+        ${this.accordionsShown.includes(rowId) ? EXPANDED_CLASS : ''}
+        `}
+        role="row"
+      >
+        {rowCells}
```

Comment:
the cell data may be normal text or some time it may be a text with link (Principal column.. text with blue color and underlined), is that condition is handled while generating the content for the row cell?

Reply:
Yes, it supports rendering any kind of custom content in Cells, nested row content, etc.

We use the concept of templates so that developers can create pre-programmed templates for custom contents, name the template, and then indicate the template name as cell, row content renderer.

`{ template: 'ticketId', payload: { id: 'NTM000021063' } }`

![Screenshot 2021-02-05 at 18 45 36](https://user-images.githubusercontent.com/25136131/107069879-d80e0780-67e2-11eb-9f9f-9efc60693b39.png)